### ISSUE TITLE
Cannot turn BUILD_WITH_URDF_SUPPORT to OFF

### ISSUE THREAD

**Issue by proyan** [stack-of-tasks/pinocchio#1159]
Hi,
URDF_SUPPORT cannot be turned off in the current devel. If I turn it off, either there is a simple cmake error in `CMakeLists.txt:L104`, `URDFDOM_VERSION doesn't exist`, or if I fix it, there are errors in `parsers/urdf/model.cpp`, given by:
```
[ 50%] Building CXX object src/CMakeFiles/pinocchio.dir/parsers/urdf/model.cpp.o
cd /local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE/src && /usr/bin/c++   -DBOOST_MPL_LIMIT_LIST_SIZE=30 -DBOOST_MPL_LIMIT_VECTOR_SIZE=30 -DPINOCCHIO_CPPAD_REQUIRES_MATRIX_BASE_PLUGIN -DPINOCCHIO_WITH_CASADI_SUPPORT -DPINOCCHIO_WITH_CPPADCG_SUPPORT -DPINOCCHIO_WITH_CPPAD_SUPPORT -DPINOCCHIO_WITH_HPP_FCL -Dpinocchio_EXPORTS -I/local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE -I/local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE/include -I/home/rbudhira/devel/src/sot/pinocchio/include -isystem /usr/include/eigen3 -isystem /local/rbudhira/devel/install/sot/include -isystem /opt/ros/kinetic/include -isystem /usr/include/assimp  -pedantic -Wno-long-long -Wall -Wextra -Wcast-align -Wcast-qual -Wformat -Wwrite-strings -Wconversion  -O3 -DNDEBUG -fPIC   -DHPP_FCL_HAVE_OCTOMAP -DFCL_HAVE_OCTOMAP -DOCTOMAP_MAJOR_VERSION=1 -DOCTOMAP_MINOR_VERSION=8 -DOCTOMAP_PATCH_VERSION=1 -o CMakeFiles/pinocchio.dir/parsers/urdf/model.cpp.o -c /home/rbudhira/devel/src/sot/pinocchio/src/parsers/urdf/model.cpp
cd /local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE/src && /usr/bin/c++   -DBOOST_MPL_LIMIT_LIST_SIZE=30 -DBOOST_MPL_LIMIT_VECTOR_SIZE=30 -DPINOCCHIO_CPPAD_REQUIRES_MATRIX_BASE_PLUGIN -DPINOCCHIO_WITH_CASADI_SUPPORT -DPINOCCHIO_WITH_CPPADCG_SUPPORT -DPINOCCHIO_WITH_CPPAD_SUPPORT -DPINOCCHIO_WITH_HPP_FCL -Dpinocchio_EXPORTS -I/local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE -I/local/rbudhira/devel/build/sot/pinocchio/_build-RELEASE/include -I/home/rbudhira/devel/src/sot/pinocchio/include -isystem /usr/include/eigen3 -isystem /local/rbudhira/devel/install/sot/include -isystem /opt/ros/kinetic/include -isystem /usr/include/assimp  -pedantic -Wno-long-long -Wall -Wextra -Wcast-align -Wcast-qual -Wformat -Wwrite-strings -Wconversion  -O3 -DNDEBUG -fPIC   -DHPP_FCL_HAVE_OCTOMAP -DFCL_HAVE_OCTOMAP -DOCTOMAP_MAJOR_VERSION=1 -DOCTOMAP_MINOR_VERSION=8 -DOCTOMAP_PATCH_VERSION=1 -o CMakeFiles/pinocchio.dir/parsers/urdf/geometry.cpp.o -c /home/rbudhira/devel/src/sot/pinocchio/src/parsers/urdf/geometry.cpp
In file included from /home/rbudhira/devel/src/sot/pinocchio/src/parsers/urdf/geometry.cpp:6:0:
/local/rbudhira/devel/install/sot/include/pinocchio/parsers/urdf/types.hpp:45:32: error: redefinition of ‘template<class T, class U> boost::shared_ptr<X> urdf::const_pointer_cast(const boost::shared_ptr<U>&)’
   PINOCCHIO_URDF_SHARED_PTR(T) const_pointer_cast(PINOCCHIO_URDF_SHARED_PTR(U) const & r)

```
Support would be welcome here. Thanks.

**Comment by proyan** [comment]
Thanks :)

**Comment by jcarpent** [comment]
Solved by #1160.
